core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -251,6 +254,10 @@ class FailedPartitions {
   def contains(topicPartition: TopicPartition): Boolean = synchronized {
     failedPartitionsSet.contains(topicPartition)
   }
+
+  def failedPartitions(): Set[TopicPartition] = synchronized {
```

Comment:
nit: Should we call it `partitions` because we already know that they are failed?